src/main/scala/devices/gpio/GPIO.scala

```diff
@@ -135,30 +135,29 @@ trait HasGPIOModuleContents extends MultiIOModule with HasRegMap {
   val rise = ~valueReg & inSyncReg;
   val fall = valueReg & ~inSyncReg;
 
-  val iofEnFields =  if (c.includeIOF) (Seq(RegField.rwReg(c.width, iofEnReg.io))) else (Seq(RegField(c.width)))
-  val iofSelFields = if (c.includeIOF) (Seq(RegField(c.width, iofSelReg))) else (Seq(RegField(c.width)))
+  val iofEnFields =  if (c.includeIOF) (Seq(RegField.rwReg(c.width, iofEnReg.io, Some(RegFieldDesc("iof_en","HW I/O functon enable", reset=Some(0)))))) else (Seq(RegField(c.width)))
+  val iofSelFields = if (c.includeIOF) (Seq(RegField(c.width, iofSelReg, RegFieldDesc("iof_sel","HW I/O function select", reset=Some(0))))) else (Seq(RegField(c.width)))
 
 
   // Note that these are out of order.
   regmap(
-    GPIOCtrlRegs.value     -> Seq(RegField.r(c.width, valueReg)),
-    GPIOCtrlRegs.output_en -> Seq(RegField.rwReg(c.width, oeReg.io)),
-    GPIOCtrlRegs.rise_ie   -> Seq(RegField(c.width, riseIeReg)),
-    GPIOCtrlRegs.rise_ip   -> Seq(RegField.w1ToClear(c.width, riseIpReg, rise)),
-    GPIOCtrlRegs.fall_ie   -> Seq(RegField(c.width, fallIeReg)),
-    GPIOCtrlRegs.fall_ip   -> Seq(RegField.w1ToClear(c.width, fallIpReg, fall)),
-    GPIOCtrlRegs.high_ie   -> Seq(RegField(c.width, highIeReg)),
-    GPIOCtrlRegs.high_ip   -> Seq(RegField.w1ToClear(c.width, highIpReg, valueReg)),
-    GPIOCtrlRegs.low_ie    -> Seq(RegField(c.width, lowIeReg)),
-    GPIOCtrlRegs.low_ip    -> Seq(RegField.w1ToClear(c.width,lowIpReg, ~valueReg)),
-    GPIOCtrlRegs.port      -> Seq(RegField(c.width, portReg)),
-    GPIOCtrlRegs.pullup_en -> Seq(RegField.rwReg(c.width, pueReg.io)),
+    GPIOCtrlRegs.value     -> Seq(RegField.r(c.width, valueReg, RegFieldDesc("value","Pin value", reset=Some(0)))),
```

Comment:
Can we make the actual names and descriptions of these registers match what they are in the document:

![screen shot 2018-04-10 at 12 32 34 pm](https://user-images.githubusercontent.com/17858596/38579292-660e598c-3cbb-11e8-9e31-dcff17a2f25d.png)

Reply:
`input_val` should be `volatile=true`